task.js

```diff
@@ -0,0 +1,231 @@
+        director: "Christopher Nolan",
+    },
+    {
+        title: "Pulp Fiction",
+        year: 1994,
+        rating: 8.9,
+        genre: ["Crime", "Drama"],
+        director: "Quentin Tarantino",
+    },
+    {
+        title: "The Matrix",
+        year: 1999,
+        rating: 8.7,
+        genre: ["Action", "Sci-Fi"],
+        director: "Lana Wachowski, Lilly Wachowski",
+    },
+    {
+        title: "Fight Club",
+        year: 1999,
+        rating: 8.8,
+        genre: ["Drama"],
+        director: "David Fincher",
+    },
+];
+
+console.log("--------------------1-------------------");
+
+//1. - Create an array of movie titles
+
+console.log(filmsInfo);
+console.log(filmsInfo.map((film) => film.title));
+console.log(filmsInfo);
+
+console.log("--------------------2-------------------");
+
+// 2. - Create an array of movies with a rating higher than 8.5. -> changed to 8.7 rating to see the differance
+
+console.log(filmsInfo);
+console.log(filmsInfo.filter((film) => film.rating > 8.7));
+console.log(filmsInfo);
+
+console.log("--------------------3-------------------");
+
+// 3. - Sort the movies by rating in descending order.
+
+console.log(filmsInfo);
+console.log(filmsInfo.sort((a, b) => (a.rating > b.rating ? -1 : 1)));
+console.log(filmsInfo);
+
+console.log("--------------------4-------------------");
+
+// 4. - Create an array of movies released after the year 2000.
+
+console.log(filmsInfo);
+console.log(filmsInfo.filter((film) => film.year > 2000));
+console.log(filmsInfo);
+
+console.log("--------------------5-------------------");
+
+// 5. - [ ] Sort the movies by year of release in ascending order.
+
+console.log(filmsInfo);
+console.log(filmsInfo.sort((a, b) => (a.year > b.year ? 1 : -1)));
+console.log(filmsInfo);
+
+console.log("--------------------6-------------------");
+
+// 6.- [ ] Create an object that counts the number of movies for each genre.
+
+const genreCount = filmsInfo.reduce((acc, film) => {
+    film.genre.forEach((genre) => {
+        acc[genre] = acc[genre] ? acc[genre] + 1 : 1;
+    });
+    return acc;
+}, {});
+
+console.log(filmsInfo);
+console.log(genreCount);
+console.log(filmsInfo);
+
+console.log("--------------------7-------------------");
+
+// 7. - [ ] Check if all movies have a rating higher than 7.
+
+console.log(filmsInfo);
+console.log(filmsInfo.every((film) => film.rating > 7));
+console.log(filmsInfo);
+
+console.log("--------------------8-------------------");
+
+// 8. - [ ] Calculate the average rating of all movies.
+
+console.log(filmsInfo);
+console.log(
+    filmsInfo.reduce((acc, film) => acc + film.rating, 0) / filmsInfo.length
+);
+console.log(filmsInfo);
+
+console.log("--------------------9-------------------");
+
+// 9. - [ ] Create an array of objects with only the title and rating of each movie.
+
+console.log(filmsInfo);
+console.log(
+    filmsInfo.map((film) => ({
+        title: film.title,
+        rating: film.rating,
+    }))
+);
```

Comment:
```suggestion
    filmsInfo.map(({ title, rating}) => ({
        title,
        rating,
    }))
);
```
You can use destructing here to reduce the function further